wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -107,6 +89,65 @@ private[bitcoins] trait TransactionProcessing extends WalletLogger {
     f
   }
 
+  /** Helper method to process a block. This fetches all of our relevent spending info dbs
+    * up front rather than fetching them every time [[processTransaction]] is called. This
+    * significantly improves performance on rescans or IBD with an existing wallet
+    */
+  private def processBlockCachedUtxos(block: Block): Future[Wallet] = {
+    //fetch all received spending info dbs relevant to txs in this block to improve performance
+    val receivedSpendingInfoDbsF =
+      spendingInfoDAO.findTxs(block.transactions.toVector)
+
+    val cachedReceivedF = receivedSpendingInfoDbsF
+
+    //fetch all spending infoDbs for this block to improve performance
+    val spentSpendingInfoDbsF =
+      spendingInfoDAO.findOutputsBeingSpent(block.transactions.toVector)
+
+    //we need to keep a cache of spentSpendingInfoDb
+    //for the case where we receive & then spend that
+    //same utxo in the same block
+    var cachedSpentF = spentSpendingInfoDbsF
+
+    val wallet = {
+      block.transactions.foldLeft(Future.successful(this)) {
+        (acc, transaction) =>
+          for {
+            _ <- acc
+            receivedSpendingInfoDbs <- cachedReceivedF
+            spentSpendingInfo <- cachedSpentF
+            processTxResult <- {
+              processTransactionImpl(
+                transaction = transaction,
+                blockHashOpt = Some(block.blockHeader.hash.flip),
+                newTags = Vector.empty,
+                receivedSpendingInfoDbsOpt = Some(receivedSpendingInfoDbs),
+                spentSpendingInfoDbsOpt = Some(spentSpendingInfo)
+              )
+            }
+            _ = {
+              //need to look if a received utxo is spent in the same block
+              //if so, we need to update our receivedSpent
+              val spentInSameBlock: Vector[SpendingInfoDb] = {
+                processTxResult.updatedIncoming.filter { spendingInfoDb =>
+                  block.transactions.exists(
+                    _.inputs.exists(
+                      _.previousOutput == spendingInfoDb.outPoint))
+                }
+              }
+
+              //add it to the cache
+              cachedSpentF =
+                Future.successful(spentSpendingInfo ++ spentInSameBlock)
```

Comment:
shouldn't this be `Future.successful(cachedSpentF ++ spentInSameBlock)`

Reply:
`cachedSpentF` is a `Future`. We need to have the underlying vector which is `<-` on above.